### DISCUSSION TITLE
Can I get a task_group into a DAG by group_id

### DISCUSSION THREAD

**Discussion by knarf20** [apache/airflow#21948] [Q&A] [answered]
Hi everyone, I have a doubt we have any method to get a task_group instance into a DAG by group_id like get_task.
Thanks in advance.

**Comment by potiuk** [comment]
Please clarify your question. It is totally unclear what you are asking about. More context is needed.

**Reply by knarf20** [reply]
For example I have the follow DAG:
```
from airflow import DAG
from airflow.utils.task_group import TaskGroup
from airflow.operators.bash import BashOperator
from datetime import datetime
with DAG('my_dag', 
schedule_interval='@daily', 
start_date=datetime(2022, 1, 1), 
catchup=False) as dag:
start = BashOperator(task_id='start', bash_command='exit 0')
with TaskGroup(group_id="processes") as processes:
task_process_a  = BashOperator(task_id='task_process_a', bash_command='exit 0')
task_process_b  = BashOperator(task_id='task_process_b', bash_command='exit 0')
end = BashOperator(task_id='end', bash_command='exit 0')
start >> processes >> end
```
and would like to do the following:

```
_task_group = dag.get_task_group(group_id=processes)
previous_tasks = _task_group.upstream_task_ids
assert previous_tasks == {"end"}
```
but I was looking through the documentation and I didn't find anything similar to a `get_task_group`.
I hope now I have explained better.

**Comment by josh-fell** [comment, marked as answer]
DAGs have access to the `TaskGroup` objects within them via the `.task_group` property. There are then a couple ways to get at the individual objects:

**Use the [`get_task_group_dict()`](https://github.com/apache/airflow/blob/5b45a78dca284e504280964951f079fca1866226/airflow/utils/task_group.py#L370) method**
```python
_task_group = dag.task_group.get_task_group_dict()["processes"]
previous_tasks = _task_group.upstream_task_ids
assert previous_tasks == {"start"}
```

**Drill down using the `.children` property for a `TaskGroup` object**
```python
_task_group = dag.task_group.children["processes"]
previous_tasks = _task_group.upstream_task_ids
assert previous_tasks == {"start"}
```
The former being a little more understandable and intuitive IMO.

**Reply by knarf20** [reply]
yes, I am just doing it that way, I just wanted to know if there was any other way and I was ignoring it, thank you very much for the answers.
